src/bp-messages/classes/class-bp-messages-message.php

```diff
@@ -343,9 +343,24 @@ public static function delete_user_message( $user_id ) {
 		$bp = buddypress();
 
 		// Get the message ids in order to delete their metas.
-		$message_ids = $wpdb->get_col( $wpdb->prepare( "SELECT DISTINCT (id) FROM {$bp->messages->table_name_messages} WHERE sender_id = %d", $user_id ) );
+		$messages_arr = BP_Messages_Message::get(
+			array(
+				'fields'  => 'ids',
+				'user_id' => $user_id,
+				'order'   => 'ASC',
```

Comment:
@janakkaneriya007 Any reason we passed order param? I can't see any order in the old query